### PR TITLE
Add support for transparent nav bar

### PR DESCRIPTION
**Pull request by natemoo-re**
- Adds a transparent style navbar
- Updates the `/db` page to use it
- Updates the `/db` page to improve hero image loading experience

### CODE REVIEW

File: src/helpers/disclosure.ts [hunk 1]
```diff
@@ -26,9 +26,17 @@ export function createDisclosure({
...
+	const footer = document.querySelector("footer")
 
 	createEffect(() => {
 		button.setAttribute("aria-expanded", String(visible()))
+		if (main) {
+			main.inert = visible()
+		}
+		if (footer) {
+			footer.inert = visible()
+		}
```
Comment (itsMapleLeaf) [inline src/helpers/disclosure.ts:39]: Doing an if check on these is a silent failure if the elements happen to change 🫠

I'd consider something like this:
```ts
	const main = expect(document.querySelector("main"))
	const footer = expect(document.querySelector("footer"))
```
```ts
// in some helpers file
export function expect<T>(value: T): NonNullable<T> {
	if (value == null) {
		throw new Error(`Expected a non-nullable value, got ${value}`)
	}
	return value
}
```

File: src/layouts/MainLayout.astro [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
 ---
 
 <!doctype html>
 <html
 	lang="en"
 	class:list={[
-		"astro-gray-700 overflow-x-hidden break-words bg-black text-astro-gray-100 [color-scheme:dark] [word-break:break-word]",
-		classList,
+		"astro-gray-700 overflow-x-hidden break-words text-astro-gray-100 [color-scheme:dark] [word-break:break-word]",
+		classList ? classList : "bg-black",
```
Comment (itsMapleLeaf) [inline src/layouts/MainLayout.astro:22]:
```suggestion
		classList || "bg-black",
```

File: src/layouts/MainLayout.astro [hunk 3]
```diff
@@ -47,6 +55,23 @@ const { class: classList, ...head } = Astro.props
...
+					left: 0;
+					z-index: 11;
+					background: #0d0f14;
+					pointer-events: none;
+					opacity: 0;
+					transition: opacity 400ms cubic-bezier(0.23, 1, 0.32, 1);
+				}
+				:root.disclosure-open body::before {
+					opacity: 1;
+				}
```
Comment (itsMapleLeaf) [inline src/layouts/MainLayout.astro:74]: This is tailwindable, but I won't block on that ¯\\\_(ツ)\_/¯

File: src/pages/db/index.astro [hunk 4]
```diff
@@ -32,7 +33,7 @@ const description = "The SQL database platform for content-driven websites."
 	</div>
 </MainLayout>
 
-<style>
+<style is:global>
 	.db {
 		background-color: #060913 !important;
 	}
```
Comment (itsMapleLeaf) [inline src/pages/db/index.astro:39]: I'd definitely rather pass this as a dynamic class name than make it global and `!important` 😂

**Review by itsMapleLeaf — APPROVED** [review]
I'll stamp this with some optional nits

File: src/helpers/disclosure.ts (hunk 1 above)
Comment (itsMapleLeaf) [reply src/helpers/disclosure.ts:39]: Also... it seems kind of scary for this to operate on global elements? It was created to mostly be a self-contained abstraction 🤔 

I also just realized that this is written in a way that `main` and `footer` might not exist on the page, so ignore the first comment I suppose! 🤪

Comment (natemoo-re) [reply src/helpers/disclosure.ts:39]: Ahh right, that's a good catch. This file is in the `src/helpers` folder, but the only place it is used is for the header... I agree that this global stuff shouldn't go here, though. I'll come up with something that makes more sense.

File: src/pages/db/index.astro (hunk 4 above)
Comment (natemoo-re) [reply src/pages/db/index.astro:39]: I swear there was a good reason to have the style block here at one point, I just totally forgot to remove it now that there's only a single style 😂

File: src/layouts/MainLayout.astro (hunk 3 above)
Comment (natemoo-re) [reply src/layouts/MainLayout.astro:74]: I anticipate that we will be having this conversation frequently 😂

Pseudo-elements feel extra strange in Tailwind, but tbh all of Tailwind feels pretty weird to me. I'll just need some time to adjust my brain.

File: src/helpers/disclosure.ts (hunk 1 above)
Comment (natemoo-re) [reply src/helpers/disclosure.ts]: Updated with an EventEmitter pattern in [394e38e](https://github.com/withastro/astro.build/pull/1033/commits/394e38ec4480a478dd3b6e54b0390452e8677e9d)

**Pull request merged**